modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -519,6 +519,49 @@
     (cond->> ((get-method sql.qp/->honeysql [:sql :relative-datetime]) driver clause)
       t (->temporal-type t))))
 
+(defmethod sql.qp/->honeysql [:bigquery-cloud-sdk :datediff]
```

Comment:
Bigquery is special case for `:week` and others fall through to `date_diff`, `datetime_diff` or `timestamp_diff`. One complication is that `timestamp_diff` does not work for year, month, or week so those fall back to datetime_diff.